usecases/telemetry/telemetry.go

```diff
@@ -0,0 +1,229 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package telemetry
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"runtime"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/entities/verbosity"
+	"github.com/weaviate/weaviate/usecases/config"
+)
+
+const (
+	defaultConsumerURL  = "https://us-central1-semi-production.cloudfunctions.net/weaviate-telemetry"
+	defaultPushInterval = 24 * time.Hour
+)
+
+type nodesStatusGetter interface {
+	LocalNodeStatus(className, output string) *models.NodeStatus
+}
+
+type modulesProvider interface {
+	GetMeta() (map[string]interface{}, error)
+}
+
+// Telemeter is responsible for managing the transmission of telemetry data
+type Telemeter struct {
+	machineID         strfmt.UUID
+	nodesStatusGetter nodesStatusGetter
+	modulesProvider   modulesProvider
+	logger            logrus.FieldLogger
+	shutdown          chan struct{}
+	failedToStart     bool
+	consumerURL       string
+	pushInterval      time.Duration
+}
+
+// Really just for testing
+type telemetryOpt func(*Telemeter)
+
+func withConsumerURL(url string) telemetryOpt {
+	return func(tel *Telemeter) {
+		tel.consumerURL = url
+	}
+}
+
+func withPushInterval(interval time.Duration) telemetryOpt {
+	return func(tel *Telemeter) {
+		tel.pushInterval = interval
+	}
+}
```

Comment:
if those just for testing , what if we just move them to testing and pass `&Telemeter` and return `&Telemeter` so that we achieve what we want without just for testing ?